api/src/main/java/org/apache/iceberg/metrics/MetricsContext.java

```diff
@@ -45,7 +46,12 @@ public String displayName() {
     }
 
     public static Unit fromDisplayName(String displayName) {
-      return Unit.valueOf(displayName.toUpperCase(Locale.ROOT));
+      Preconditions.checkArgument(null != displayName, "Invalid unit: null");
+      try {
+        return Unit.valueOf(displayName.toUpperCase(Locale.ENGLISH));
```

Comment:
I think it should be ok to change this here and in other places from `ROOT` to `ENGLISH`, since we know those will be all english words